source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,107 @@ Insert or Update in a Single Operation
 
 .. default-domain:: mongodb
 
-If your application stores and modifies data in MongoDB, you probably use
-insert and update operations. In certain workflows, you may need to choose
-between an insert and update depending on whether the document exists.
-In these cases, you can streamline your application logic by using the
-``upsert`` option available in the following methods:
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. _upsert_golang:
+
+Overview
+--------
+
+In this guide, you can learn how to specify an upsert.
+
+Sample Data
+~~~~~~~~~~~
+
+To run the example in this guide, load the sample data into the
+``ratings`` collection of the ``tea`` database with the following
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/upsert.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+.. include:: /includes/fundamentals/automatic-db-coll-creation.rst
+
+Each document contains a rating for a type of tea that corresponds to
+the ``type`` and ``rating`` fields.
+
+Upsert
+------
+
+Applications use insert and update operations to store and modify data.
+Sometimes, you need to choose between an insert and update depending on
```

Comment:
"Sometimes, you need to choose between an insert and update" reads a bit off. Perhaps monospacing insert and update would make it clearer that it is a technical term. Alternatively, perhaps you could rephrase to "Sometimes, you need to choose between an insert and an update operation depending ..."